autoprotocol/protocol.py

```diff
@@ -282,36 +261,36 @@ def ref(
             raise RuntimeError(
                 "Two containers within the same protocol cannot have the same " "name."
             )
-        opts = {}
+        opts: RefOpts = RefOpts()
 
         # Check container type
         try:
             cont_type = self.container_type(cont_type)
             if id and cont_type:
-                opts["id"] = id
+                opts.id = id
             elif cont_type:
-                opts["new"] = cont_type.shortname
+                opts.new = cont_type.shortname
         except ValueError as e:
             raise RuntimeError(
                 f"{cont_type} is not a recognized container type."
             ) from e
 
         if storage:
-            opts["store"] = {"where": storage}
+            opts.store = StorageLocation(**{"where": storage})
```

Comment:
Storage as [Location enum](https://github.com/autoprotocol/autoprotocol-python/blob/stdv-1949-update-ap-python-protocol-instruction-arguments/autoprotocol/types/ref.py#L10-L16) inside types/ref.py > VERSUS Storage as [regular String](https://github.com/autoprotocol/autoprotocol-python/blob/stdv-1949-update-ap-python-protocol-instruction-arguments/autoprotocol/protocol.py#L171) inside  protocol.py > ref() : 

I'm curious if this difference between the input params could cause breakage because currently the protocol.py > ref() function allows user to pass in any String value for storage param; but this line of code for RefOpts datatype expect specificaly for Location enum ? 

While all of ApPy tests passed given this codebase condition, I wonder if we can add new test case for protocol.ref() where the Storage input would be any String NOT existing in Location enum, and anticipate error/success?

Reply:
NOTE: @sabrinamohabbat  wrote the test in question where we expected test to fail given invalid storage value = "hot" (not applicable in Location enum). But the test unexpectedly pass (success). Abandon the test request for the moment; and I will need separate study why Python would enable not valid String in the place of enum in the method signature.